src/infrastructure/scryfall.rs

```diff
@@ -0,0 +1,117 @@
+use anyhow::anyhow;
+use serde::Deserialize;
+use uuid::Uuid;
+
+use crate::{
+    card::Card, infrastructure::InfrastructureError, infrastructure::InfrastructureResult,
+};
+
+const BASE_URL: &str = "https://api.scryfall.com/";
+
+#[derive(Deserialize)]
+#[serde(tag = "object")]
+enum ScryfallObject {
+    #[serde(rename = "list")]
+    List { data: Vec<ScryfallCard> },
+    #[serde(rename = "error")]
+    Error { details: String },
+}
+
+#[derive(Deserialize)]
+struct ScryfallCard {
+    id: Uuid,
+    oracle_id: Uuid,
+    name: String,
+    lang: String,
+    scryfall_uri: String,
+    type_line: String,
+    scryfall_set_uri: String,
+    #[serde(flatten)]
+    card_face_kind: ScryfallCardFaceKind,
+}
+
+#[derive(Deserialize)]
+enum ScryfallCardFaceKind {
+    #[serde(rename = "image_uris")]
+    SingleFace(ScryfallImageUris),
+    #[serde(rename = "card_faces")]
+    MultipleFace([ScryfallCardFace; 2]),
+}
+
+#[derive(Deserialize)]
+struct ScryfallImageUris {
+    png: String,
+}
+
+#[derive(Deserialize)]
+struct ScryfallCardFace {
+    image_uris: ScryfallImageUris,
+}
+
+#[derive(Clone)]
+pub struct ScryfallCardSearchEngine {
```

Comment:
design - 
`ScryfallCardSearchEngine` may be more generic, omitting the `Card` part may yield a more open point of view of its responsibility. this struct represents in its essence the DAL with Scryfall, and as such should handle all communication with it, not just `Card`s

Reply:
Fixed.